pkg/controller/codestatus/expire.go

```diff
@@ -51,3 +52,46 @@ func (c *Controller) HandleExpire() http.Handler {
 			})
 	})
 }
+
+func (c *Controller) HandleExpirePage() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		var code *database.VerificationCode
```

Comment:
This should be `var code database.VerificationCode` and then pass `&code` around. Otherwise you risk a NPE.